### PR TITLE
ci: clean up `circleci/config.yml`

### PR DESCRIPTION
**Pull request by gkalpak**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] ~~Tests for the changes have been added (for bug fixes / features)~~
- [ ] ~~Docs have been added / updated (for bug fixes / features)~~

## PR Type
```
[x] CI related changes
```

## What is the current behavior?
Several values to keep in sync throughout `circleci/config.yml`.

## What is the new behavior?
All values to keep in sync in one place in `circleci/config.yml`.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,21 +7,25 @@
...
-    - image: angular/ngcontainer:0.0.8
-    ## IMPORTANT
+    - image: *docker_image
 
 # After checkout, rebase on top of master.
 # Similar to travis behavior, but not quite the same.
 # See https://discuss.circleci.com/t/1662
 anchor_2: &post_checkout
-  post: git pull --ff-only origin "refs/pull/${CI_PULL_REQUEST//*pull\//}/merge"
+  post: git pull --ff-only origin "refs/pull/${CIRCLE_PULL_REQUEST//*pull\//}/merge"
```
Comment (gkalpak) [inline .circleci/config.yml]: `CI_PULL_REQUEST` is an alias for `CIRCLE_PULL_REQUEST` for backwards compatibility ([source][1]).
Switching to the new thing :smiley: 

[1]: https://circleci.com/docs/2.0/env-vars/#circleci-environment-variable-descriptions

File: .circleci/config.yml [hunk 2]
```diff
@@ -7,21 +7,25 @@
 # To validate changes, use an online parser, eg.
 # http://yaml-online-parser.appspot.com/
 
+# Variables
+
+## IMPORTANT
+# If you change the `docker_image` version, also change the `cache_key` suffix and the version of
+# `com_github_bazelbuild_buildtools` in the `/WORKSPACE` file.
+var_1: &docker_image angular/ngcontainer:0.0.8
+var_2: &cache_key angular-{{ .Branch }}-{{ checksum "yarn.lock" }}-0.0.8
```
Comment (alexeagle) [inline .circleci/config.yml]: nice, thanks! We were searching for some formulation for code sharing, but I didn't think to just put the entire cache key in an anchor.

**Review by alexeagle — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (gkalpak) [reply .circleci/config.yml]: 👍
(Too bad YAML or CircleCI do not support interpolation of arbitrary variables in string values 😞)

Comment (IgorMinar) [reply .circleci/config.yml]: sweet!

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**